src/EditorFeatures/CSharp/Formatting/CSharpEditorFormattingService.cs

```diff
@@ -173,6 +201,8 @@ private static bool TokenShouldNotFormatOnReturn(SyntaxToken token)
 
         public async Task<IList<TextChange>> GetFormattingChangesAsync(Document document, char typedChar, int caretPosition, CancellationToken cancellationToken)
         {
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Why move this here if it's not used until much later? There's a lot of checks that might bail out before we would even need the options.